metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py

```diff
@@ -59,6 +62,44 @@ class SnowflakeCommonMixin:
 
     platform = "snowflake"
 
+    @staticmethod
+    @lru_cache(maxsize=128)
+    def create_snowsight_base_url(account_id: str) -> Optional[str]:
+        cloud: Optional[str] = None
+        account_locator: Optional[str] = None
+        cloud_region_id: Optional[str] = None
+
+        if "." not in account_id:  # e.g. xy12345
+            account_locator = account_id.lower()
+            cloud_region_id = "us-west-2"
+        else:
+            parts = account_id.split(".")
+            if len(parts) == 2:  # e.g. xy12345.us-east-1
+                account_locator = parts[0].lower()
+                cloud_region_id = parts[1].lower()
+            elif len(parts) == 3 and cloud in ("aws", "gcp", "azure"):
+                # e.g. xy12345.ap-south-1.aws or xy12345.us-central1.gcp or xy12345.west-us-2.azure
+                # NOT xy12345.us-west-2.privatelink or xy12345.eu-central-1.privatelink
+                account_locator = parts[0].lower()
+                cloud_region_id = parts[1].lower()
+                cloud = parts[2].lower()
+            else:
+                logger.warning(
+                    f"Could not create Snowsight base url for account {account_id}."
+                    "This may happen if the account is using private connectivity to the Snowflake service."
+                )
+                return None
+
+        # TODO: BASE URL for Snowsight using private connectivity to the Snowflake service
+        # https://docs.snowflake.com/en/user-guide/ui-snowsight-gs.html#signing-in-to-web-interface
+        # alternatively, base_url can be taken as part of config to support this.
+        if cloud is None or cloud == "aws":
```

Comment:
Can we add some constants here? For the cloud identifier.

SNOWFLAKE_CLOUD_ID_AWS = "aws"
SNOWFLAKE_CLOUD_ID_GCP = "gcp"

etc. You could also do an enum.

Let's avoid these raw-string comparisons.